src/test/integration_tests/api/s3/test_s3_bucket_policy_iam_user.js

```diff
@@ -0,0 +1,488 @@
+
+        await rpc_client.account.delete_account({ email: user_a });
+        await rpc_client.account.delete_account({ email: user_b });
+    });
+
+    mocha.it('IAM User access bucket with Bucket policy', async function() {
+        if (is_nc_coretest) this.skip(); // eslint-disable-line no-invalid-this
+        const input = {
+            UserName: user_a
+        };
+        // 1. Create IAM user account
+        const command = new CreateUserCommand(input);
+        const response = await iam_account_a.send(command);
+        _check_status_code_ok(response);
+        assert.equal(response.User.UserName, user_a);
+        user_a_arn = response.User.Arn;
+        user_a_id = response.User.UserId;
+        // 2. Create bucket policy with that IAM user and add to bucket
+        const s3_policy = {
+            Version: '2012-10-17',
+            Statement: [
+                {
+                    Action: ['s3:PutObject'],
+                    Effect: 'Allow',
+                    Principal: { AWS: [user_a_arn] },
+                    Resource: [`arn:aws:s3:::${BKT}/*`],
+                }
+            ]};
+
+        await s3_owner.putBucketPolicy({
+            Bucket: BKT,
+            Policy: JSON.stringify(s3_policy)
+        });
+        const res_get_bucket_policy = await s3_owner.getBucketPolicy({
+            Bucket: BKT,
+        });
+        assert.equal(res_get_bucket_policy.$metadata.httpStatusCode, 200);
+        // 3. Create access and secret key for the IAM user
+        const access_command = new CreateAccessKeyCommand(input);
+        const access_response = await iam_account_a.send(access_command);
+
+        const access_key_id = access_response.AccessKey.AccessKeyId;
+        const secret_key = access_response.AccessKey.SecretAccessKey;
+        assert(access_key_id !== undefined);
+
+        // 4. Add inline policy to account, without inline policy S3 access will fail.
+        const inline_input = {
+            UserName: user_a,
+            PolicyName: policy_name,
+            PolicyDocument: allow_all_iam_user_inline_policy_document
+        };
+        const inline_command = new PutUserPolicyCommand(inline_input);
+        const inline_response = await iam_account_a.send(inline_command);
+        _check_status_code_ok(inline_response);
+
+        // 5. Try to put object to bucket with IAM user s3 client, Should not fail
+        iam_user_a_s3_creds.credentials = {
+            accessKeyId: access_key_id,
+            secretAccessKey: secret_key,
+        };
+        s3_user_a = new S3(iam_user_a_s3_creds);
+        const res_put_object = await s3_user_a.putObject({
+            Body: BODY,
+            Bucket: BKT,
+            Key: KEY,
+        });
+        assert.equal(res_put_object.$metadata.httpStatusCode, 200);
+        await s3_owner.deleteObject({
+            Bucket: BKT,
+            Key: KEY
+        });
+    });
+
+    mocha.it('IAM User access bucket that its owner account owns', async function() {
```

Comment:
Perhaps write explicitly that there is no bucket policy?
```suggestion
    mocha.it('IAM User access bucket that its owner account owns (no bucket policy)', async function() {
```